x-pack/legacy/plugins/apm/public/components/app/ServiceMap/EmptyBanner.tsx

```diff
@@ -7,37 +7,70 @@
 import { EuiCallOut } from '@elastic/eui';
 import lightTheme from '@elastic/eui/dist/eui_theme_light.json';
 import { i18n } from '@kbn/i18n';
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import styled from 'styled-components';
 import { ElasticDocsLink } from '../../shared/Links/ElasticDocsLink';
+import { CytoscapeContext } from './Cytoscape';
 
-const EmptyBannerCallOut = styled(EuiCallOut)`
+const EmptyBannerContainer = styled('div')`
   margin: ${lightTheme.gutterTypes.gutterSmall};
   /* Add some extra margin so it displays to the right of the controls. */
-  margin-left: calc(
-    ${lightTheme.gutterTypes.gutterLarge} +
-      ${lightTheme.gutterTypes.gutterExtraLarge}
+  left: calc(
+    ${lightTheme.gutterTypes.gutterExtraLarge} +
+      ${lightTheme.gutterTypes.gutterSmall}
   );
   position: absolute;
   z-index: 1;
 `;
 
 export function EmptyBanner() {
+  const cy = useContext(CytoscapeContext);
+  const [nodeCount, setNodeCount] = useState(0);
+
+  useEffect(() => {
+    const handler: cytoscape.EventHandler = event =>
+      setNodeCount(event.cy.nodes().length);
+
+    if (cy) {
+      cy.on('add remove', 'node', handler);
+    }
+
+    return () => {
+      if (cy) {
+        cy.removeListener('add remove', 'node', handler);
+      }
+    };
+  }, [cy]);
+
+  // Only show if there's a single node.
+  if (!cy || nodeCount !== 1) {
+    return null;
+  }
+
+  // Since we're absolutely positioned, we need to get the full width and
+  // subtract the space for controls and margins.
+  const width =
+    cy.width() -
+    parseInt(lightTheme.gutterTypes.gutterExtraLarge, 10) -
+    parseInt(lightTheme.gutterTypes.gutterLarge, 10);
```

Comment:
If the user resizes the window, does this width get updated?

Reply:
No. We could observe the cytoscape resize event or the window event but I don't think it's worth the extra complexity.